### PR TITLE
feat: add lazy loading feature to class "Commit"

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2jact3q 将 commit_id + name 作为 Commit 生成的必要条件，其他属性设置为懒加载](https://app.clickup.com/t/2jact3q)

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/commit.py [hunk 1]
```diff
@@ -129,8 +133,65 @@ def to_pyobj(self) -> Dict[str, str]:
                     "committed_at": <str>
                 }
 
+        Returns:
+            A :class:`Commit` instance created from the input python dict.
+
+        """
+        obj: _C = object.__new__(cls)
```
Comment (linjiX) [inline graviti/manager/commit.py:140]:
```python
obj = cls(dataset, contents["commit_id"])
```

File: graviti/manager/commit.py [hunk 2]
```diff
@@ -44,36 +46,36 @@ class Commit(Sheets, AttrsMixin):
 
     """
 
-    _repr_attrs: Tuple[str, ...] = ("parent_commit_id", "title", "committer", "committed_at")
+    _commit_info: Dict[str, Optional[str]]
```
Comment (linjiX) [inline graviti/manager/commit.py:49]: shall we put `_repr_attrs` above `_commit_info` due to `_repr_attrs` is a class variable?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/commit.py [hunk 3]
```diff
@@ -209,50 +263,31 @@ class NamedCommit(Commit):  # pylint: disable=too-many-instance-attributes
...
-        self.commit_id = commit_id
-        if self.commit_id is None:
-            return
-
-        self.parent_commit_id = parent_commit_id
-        self.title = title
-        self.description = description
-        self.committer = committer
-        self.committed_at = committed_at
+        self.commit_id = commit_id  # type: ignore[assignment]
```
Comment (linjiX) [inline graviti/manager/commit.py:279]: shall we call `super().__init__` here?

Comment (linjiX) [reply graviti/manager/commit.py:279]: also where the ignore[assignment] comes from?

File: graviti/manager/commit.py [hunk 4]
```diff
@@ -264,8 +299,13 @@ def to_pyobj(self) -> Dict[str, str]:
                     "committed_at": <str>
                 }
 
+        Returns:
+            A :class:`NamedCommit` instance created from the input python dict.
+
         """
-        return self._dumps()
+        obj = super().from_response(dataset, contents)
```
Comment (linjiX) [inline graviti/manager/commit.py:306]: shall we call `__init__` here?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/commit.py [hunk 5]
```diff
@@ -209,50 +268,30 @@ class NamedCommit(Commit):  # pylint: disable=too-many-instance-attributes
...
         name: str,
         commit_id: Optional[str],
-        parent_commit_id: Optional[str],
-        title: str,
-        description: str,
-        committer: str,
-        committed_at: str,
     ) -> None:
-        self._dataset = dataset
+        super().__init__(dataset, commit_id)  # type: ignore[arg-type]
```
Comment (linjiX) [inline graviti/manager/commit.py:282]: where does this ignore come from?

**Comment by linjiX** [comment]
commit type should be `feat`

**Review by linjiX — APPROVED** [review]

**Pull request merged**